test/cql-pytest/cassandra_tests/validation/operations/compact_storage_test.py

```diff
@@ -12,6 +12,12 @@
 
 compactOption = " WITH COMPACT STORAGE"
 
+# Enable compact_storage for all tests
+@pytest.fixture(scope="module")
+def all_tests_compact_storage(compact_storage):
+    pass
```

Comment:
I am not happy at all about this approach. It took me a while to understand why this actually works: The "compact_storage" fixture turns on compact storage for the ENTIRE SERVER, and never disables it, so if you run a test with this fixture *first*, all other tests will work. But if pytest reorders the tests, or we do (I think @xtrey is working on doing just that) something like dtest where we deliberately run just a subset of the test functions, this will not work as you hoped it will.

I think you should either use a function-scoped fixture (as in the previous version), or just change the configuration in run.py and test/pytest/scylla_cluster.py. We already do the latter for other deprecated features we want to keep testing and experimental features we want to test - we need to enable them otherwise we can't test them.

Reply:
Test.py triggers the pytest under the hood by file. Pytest by itself does not guarantee the order of the executed tests. So even if it works right now, it's fragile. I totally share the @nyh concerns about this.
However, there is pytest [hook](https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_collection_modifyitems) where we can modify the order of the tests and put one of the tests as a first one. So we can ensure it will always execute in a first row.

Reply:
I don't mind solving this issue in a better way that won't require using hooks

Reply:
But I'd still like to enable compact storage only for functions or complete modules that require it